### DISCUSSION TITLE
[PLUGIN] Expose tailwind classes dynamically created by JavaScript?

### DISCUSSION THREAD

**Discussion by sveltecult** [tailwindlabs/tailwindcss#11755] [Help] [answered]
Hello, I'm developing (aka porting) a plugin and there are classes that are not on the markup but added by JS instead. So, tailwind can't see it and therefore, won't add to the css in JIT. I want certain classes exposed even if it's not used in the markup. How can I do that?

A use case would be a class `foo` is added by JS in the <body> tag but that my markup does not have `<body class="foo">` so tailwind can't compile it because it's non-existent. I want that exposed by default even if it's unused.

Well, we could just put that to the `app.css` file but would be nice if I can do it in a plugin level so I don't need to instruct users to manually put that.

I tried adding the classes to `addBase()` but that did not add the class I want or I did something wrong.

Any help would be greatly appreciated. Thanks.

EDIT: If you could also show how do I import css inside a plugin. Thank you. :pray:

**Comment by sveltecult** [comment, marked as answer]
Ok solved it. For those who are also wondering, it's "safelist" in the tailwind.config.js :)

**Reply by mybearworld** [reply]
You could also add your JS file to the content.
